### PR TITLE
feat(btc): add BTC account creation menu entry

### PR DESCRIPTION
**Pull request by ccharly**
## **Description**

This adds a new menu entry "Create a new Bitcoin account" to create a Bitcoin account using the Bitcoin Manager Snap.

> [!NOTE]
> This new feature is only available on Flask for now.

> [!WARNING]
> Some code logic of this PR might change depending if we are able to reduce the e2e test flakiness from this PR:
> - https://github.com/MetaMask/metamask-extension/pull/25191

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25625?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/accounts-planning/issues/451

### Main build

1. `yarn start`
2. Click on the account (in the app header)
3. Click "Add account or hardware wallet"
3. **You should not see anything related to Bitcoin account creation**

Screenshot 2024-07-02 at 11 45 22

### Flask build

1. `yarn start`
2. Click on the account (in the app header)
3. Click "Add account or hardware wallet"
4. Click "Create a new Bitcoin account"
5. Input your account name
6. You should now see your new Bitcoin account in the extension

Screenshot 2024-07-02 at 11 53 17
Screenshot 2024-07-02 at 11 54 12
Screenshot 2024-07-02 at 11 54 25
Screenshot 2024-07-02 at 11 54 31

### **After**

Screenshot 2024-07-02 at 11 45 22
Screenshot 2024-07-02 at 11 53 17
Screenshot 2024-07-02 at 11 54 12
Screenshot 2024-07-02 at 11 54 25
Screenshot 2024-07-02 at 11 54 31

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/create-btc-account/create-btc-account.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    if (name) {
+      // READ THIS CAREFULLY:
+      // We have to update the redux state here, since we are updating the global state
+      // from the background during account creation
+      await forceUpdateMetamaskState(dispatch);
+
+      // NOTE: If the renaming part of this flow fail, the account might still be created, but it
+      // will be named according the Snap keyring naming logic (Snap Account N).
+      dispatch(setAccountLabel(account.address, name));
+    }
```
Comment (montelaidev) [inline ui/components/multichain/create-btc-account/create-btc-account.tsx:42]: maybe we should extract this out into an action

**Comment by montelaidev** [comment]
Tested and the account creation with the preinstalled flask is working. The only issue is with the asset icon
Screenshot 2024-07-09 at 17 33 02

File: ui/components/multichain/create-btc-account/create-btc-account.tsx (hunk 1 above)
Comment (ccharly) [reply ui/components/multichain/create-btc-account/create-btc-account.tsx]: I kinda followed the same pattern than for `create-eth-account`, so maybe we should refactor them all at once?

**Comment by ccharly** [comment]
> Tested and the account creation with the preinstalled flask is working. The only issue is with the asset icon

@montelaidev Yes the asset icon is not showing during balance's loading (same behavior than for eth actually)

**Comment by gantunesr** [comment]
@ccharly - If the account was already created, the button "Add new Bitcoin account" should be disabled. Once the account is deleted/removed, the button should be enabled again

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -278,7 +278,10 @@
     "description": "$1 is Learn more link"
   },
   "addNewAccount": {
-    "message": "Add a new account"
+    "message": "Add a new Ethereum account"
+  },
+  "addNewBitcoinAccount": {
+    "message": "Add a new Bitcoin account"
```
Comment (danroc) [inline app/_locales/en/messages.json:284]: According to the copy it should have a `(Beta)` suffix.

```suggestion
    "message": "Add a new Bitcoin account (Beta)"
```

**Comment by ccharly** [comment]
> @ccharly - If the account was already created, the button "Add new Bitcoin account" should be disabled. Once the account is deleted/removed, the button should be enabled again

@gantunesr yup, just implemented it here: [5a196a0](https://github.com/MetaMask/metamask-extension/pull/25625/commits/5a196a0ef963ef6844def1650f9c5dc62336ff4d)

But I had to wait for other PRs to be merged! :)

**Review by danroc — APPROVED** [review]

**Comment by montelaidev** [comment]
Tested again, and the menu entry is now disabled after the account creation.

**Review by montelaidev — APPROVED** [review]

**Pull request merged**